elastic/kibana
# [Index Management] Be more explicit about data retention values

Closes: https://github.com/elastic/kibana/issues/167475

## Summary
This PR adds expands the data retention values in the DS table and details flyout. So that instead of being `2s` we would show `2 seconds` , for example.

##### How to test
1. Start elasticsearch with `yarn es snapshot --license=trial -E path.repo=/tmp/snapshots` and kibana with `yarn start`
2. Execute console request to generate the DS (the last command can be edited to change the values of the data retention)
3. Navigate to index management and click on data streams tab 
4. Verify that:
  * The data retention unit in the table is expanded
  * The data retention unit in the flyout is expanded

<details>
<summary>Console request for generating streams</summary>

```
PUT _cluster/settings
{
  "persistent": {
    "indices.lifecycle.poll_interval": "1s",
    "data_streams.lifecycle.poll_interval": "1s",
    "cluster.lifecycle.default.rollover": "min_docs=2,max_docs=2"
  }
}

PUT _ilm/policy/rollover-w-ilm
{
  "policy": {
    "phases": {
      "hot": {
        "actions": {
          "rollover": {
            "max_docs": "1"
          }
        }
      }
    }
  }
}

PUT _index_template/my-data-stream-template
{
  "index_patterns": ["my-data-stream*"],
  "data_stream": {},
  "priority": "400",
  "template": {
    "settings": {
      "index.lifecycle.name": "rollover-w-ilm"
    }
  }
}

POST my-data-stream/_doc?refresh
{
  "@timestamp": "2023-05-06T16:21:15.000Z",
  "sensor_id": "SKYNET-000001",
  "location": "swamp"
}



PUT _index_template/my-data-stream-template
{
  "index_patterns": ["my-data-stream*"],
  "data_stream": {},
  "priority": "400",
  "template": {
    "settings": {
      "index.lifecycle.name": "rollover-w-ilm",
      "index.lifecycle.prefer_ilm": false
    },
    "lifecycle": {
      "data_retention": "7d"
    }
  }
}

POST my-data-stream2/_doc?refresh
{
  "@timestamp": "2023-05-06T16:21:15.000Z",
  "sensor_id": "SKYNET-000001",
  "location": "swamp"
}

PUT _data_stream/my-data-stream/_lifecycle
{
  "data_retention": "7m"
}
```
</details>

<details>
<summary>Screenshots</summary>

![Screenshot 2023-11-01 at 10 00 54](https://github.com/elastic/kibana/assets/1191206/76321c97-ee8c-4da6-ad3b-b2ca5bb64db1)
![Screenshot 2023-11-01 at 09 51 11](https://github.com/elastic/kibana/assets/1191206/c7113b96-0322-49c3-b44e-a68f32327985)
![Screenshot 2023-11-01 at 09 51 40](https://github.com/elastic/kibana/assets/1191206/e3c50142-af87-46ee-b848-8455eb37d961)

</details>
